### DISCUSSION TITLE
Can autogluon fit and predict with sliding windows?

### DISCUSSION THREAD

**Discussion by liyiersan** [autogluon/autogluon#3954] [Q&A] [answered]
Thanks for your great work! I wonder if it would be possible to use autogluon for very long-term time series forecasting with sliding windows.

For example, I have time series data for 2020-2021, which are collected every 15 minutes. Therefore, I get $4 \times 24 \times 365 \times 2 = 70,080$ rows in total. If I would like to predict the data in 2022, would it be ok if I directly set the prediction length to $4 \times 24 \times 365 = 35,040$ 

Are the following codes correct?

```python
import pandas as pd
from autogluon.timeseries import TimeSeriesDataFrame, TimeSeriesPredictor

def load_csv_data(cvs_file_path):
    df = pd.read_csv(cvs_file_path, na_values='<NULL>')
    # replace NULL with previous value
    df.ffill(inplace=True)

    # convert dateformat to datetime
    df['timestamp'] = pd.to_datetime(df['timestamp'])
    
	# reoder the columns
    data_cols = df.columns[2:-1] # exclude item_id, date, and target columns
    df = df[['item_id', 'timestamp', 'target'] + [col for col in data_cols]]

    tsdf = TimeSeriesDataFrame(df, id_column='item_id', timestamp_column='timestamp')
    return tsdf
    
train_path = './data/train.csv'
test_path = './data/test.csv'
train_tsdf = load_csv_data(train_path)
predictor = train_model(train_tsdf)
predictor = TimeSeriesPredictor(prediction_length=35040, eval_metric='RMSE',
                                known_covariates_names=train_tsdf.columns[1:])
test_tsdf = load_csv_data(test_path)
predictions = predictor.predict(train_tsdf, known_covariates=test_tsdf)
```

The above codes may cause the out-of-memory issue, can I set a contextual length for time series forecasting? For instance, using one-month data to predict next-week data. To do so, what should I do? Can autogluon fit and predict with sliding windows?

**Comment by liyiersan** [comment]
I have carefully read the source code of deep learning models. There are parameters like " context_length,  batch_size, predict_batch_size" to restrict the GPU memory usage, I wonder why it could cause OOM on a 24GB GPU.

**Reply by liyiersan** [reply]
Problems solved, someone else is using the GPU.

**Comment by shchur** [comment, marked as answer]
Hi, thanks for opening the issue. The code that you shared looks mostly correct, one small detail that is missing is the call to `predictor.fit()`. 

In principle, the code should work but we haven't really benchmarked AutoGluon with such long forecast horizons at such a fine granularity. It is possible that some models will become extremely slow in this case (e.g., AutoETS or AutoARIMA). If that happens, you can exclude the slow models using the `excluded_model_types` argument to fit, e.g. `predictor.fit(..., excluded_model_types=["AutoETS", "AutoARIMA"])`. 

If you encounter issues with some specific models (e.g., model DeepAR is crashing with an out-of-memory error), please open an issue for that + provide logs and other information necessary to reproduce the problem.
